packages/app/pages/Organization/organizationCatalog.tsx

```diff
@@ -0,0 +1,201 @@
+// React component for the frontside of the card
+import * as React from "react";
+import * as themable from "react-themeable";
+import CardFront from "./orgCard";
+import * as pageTheme from "./catalogStyle.scss";
+import {sortByName, sortByDistance, getDistanceBetweenPoints, filterByType, filterByDistance} from "./utilities"
+import getAllOrganizations from "../../../ui-bindings/Organization/getAllOrganizations";
+import {isNullOrUndefined} from "util";
+
+
+class SearchResults extends React.Component {
+
+  constructor() {
+    super();
+    this.state = {
+      sorting: "alphabetical",
+      typeFilter: "All",
+      distanceFilter: 50,
+      load: true,
+      nameFilter: "",
+      myOrgsFilter: false,
+      pubOrgsFilter: true
+    };
+  }
+
+  onSortFilterChange = function (value) {
+    this.setState({sorting: value});
+  };
+
+  onTypeFilterChange = function (value) {
+    this.setState({typeFilter: value});
+  };
+
+  onDistanceFilterChange = function (event, override) {
+    if (event.key === 'Enter' || override) {
+      this.setState({distanceFilter: event.target.value});
+    }
+  };
+
+  handleReset = function (env) {
+    env.setState({
+      distanceFilter: 50,
+      sorting: "alphabetical",
+      typeFilter: "All",
+      nameFilter: "",
+      myOrgsFilter: false,
+      pubOrgsFilter: true
+    });
+
+  };
+  onNameChange = function (event) {
+    if (event.key === 'Enter') {
+      this.setState({nameFilter: event.target.value});
+    }
+  };
+  onMyOrgsChange = function (value) {
+    this.setState({myOrgsFilter: value});
+  };
+  onPubOrgsChange = function (value) {
+    this.setState({pubOrgsFilter: value});
+  };
+
+  notImplementedFunction = function () {
+    alert("This feature is not yet implemented");
+  };
+
+
+  render() {
+    let page_theme = themable(pageTheme);
+    let msoeCC = {latitude: 43.044004, longitude: -87.909020};
+    const OrgList = getAllOrganizations(({organizationList, loading, error}) => {
+
+      if (loading) {
+        return (
+          <strong>Loading...</strong>
+        );
+      } else if (error) {
+        return (<p style={{color: "#F00"}}>API error</p>);
+      } else {
+        let filteredOrgs = [];
+
+        for (let org of organizationList) {
```

Comment:
Should we move this filtering code to a method above? This can help address some of the concerns you brought up about the size of this return method @StrikeNP